mslib/mscolab/file_manager.py

```diff
@@ -250,6 +256,58 @@ def modify_user(self, user, attribute=None, value=None, action=None):
             db.session.commit()
         return True
 
+    def upload_file(self, file, subfolder, identifier=None):
+        """
+        Generic function to save files securely in specified directory with unique filename
+        and return the relative file path.
+        """
+        upload_folder = mscolab_settings.UPLOAD_FOLDER
+        with fs.open_fs(upload_folder) as _fs:
+            file_dir = fs.path.join(upload_folder, str(subfolder) if subfolder else "")
+            if sys.platform.startswith('win'):
+                file_dir = file_dir.replace('\\', '/')
+            _fs.makedirs(str(subfolder) if subfolder else "", recreate=True)
+
+            # Creating unique and secure filename
+            file_name, _ = file.filename.rsplit('.', 1)
+            mime_type, _ = mimetypes.guess_type(file.filename)
+            file_ext = mimetypes.guess_extension(mime_type) if mime_type else '.unknown'
+            token = secrets.token_urlsafe()
+            timestamp = time.strftime("%Y%m%dT%H%M%S")
+
+            if identifier:
+                file_name = f'{identifier}-{timestamp}-{token}{file_ext}'
+            else:
+                file_name = f'{file_name}-{timestamp}-{token}{file_ext}'
+            file_name = secure_filename(file_name)
+
+            # Saving the file
+            file_path = fs.path.join(file_dir, file_name)
+            file.save(file_path)
+
+            # Relative File path
+            upload_folder = upload_folder.replace('\\', '/')
+            static_dir = fs.path.basename(upload_folder)
+            static_file_path = os.path.join(static_dir, str(subfolder), file_name)
+            static_file_path = static_file_path.replace('\\', '/')
+            logging.debug(f'Relative Path: {static_file_path}')
+
+            return static_file_path
+
+    def save_user_profile_image(self, user_id, image_file):
+        """
+        Save the user's profile image path to the database.
+        """
+        relative_file_path = self.upload_file(image_file, subfolder='profile', identifier=user_id)
```

Comment:
This is hardcoding the "profile" directory as the subfolder under the `UPLOAD_FOLDER`. I think this would be fine, but then we can not also have `PROFILE_IMG_FOLDER` as a config option. It must be consistent.